hack/make/validate-vendor

```diff
@@ -0,0 +1,33 @@
+#!/bin/bash
+
+source "${MAKEDIR}/.validate"
+
+IFS=$'\n'
+files=( $(validate_diff --diff-filter=ACMR --name-only -- 'hack/vendor.sh' || true) )
```

Comment:
We probably want to adjust this so that it includes `vendor/` as well, and then change the `for` loop below to instead be something like `if [ "${#files[@]}" -gt 0 ]; then` so that it runs if either `hack/vendor.sh` or anything in the `vendor` directory have changed and verifies both.